eip/EIPS/eip-4337.md

```diff
@@ -68,35 +79,55 @@ function simulateValidation
 
 The core interface required for a wallet to have is:
 
-```c++
-function validateUserOp
-    (UserOperation calldata userOp, bytes32 requestId, uint256 missingWalletFunds)
-    external;
+```solidity
+interface IWallet {
+  function validateUserOp
+      (UserOperation calldata userOp, bytes32 requestId, uint256 missingWalletFunds)
+      external;
+}
 ```
+The wallet
+* MUST validate the caller is a trusted EntryPoint
+* The requestId is a hash over the userOp (except signature), entryPoint and chainId
+* MUST validate the signature is a valid signature of the `requestId`
+* MUST pay the entryPoint (caller) at least the "missingWalletFunds" 
+* the wallet MAY pay more than this minimum, to cover future transactions
+* Note that in case the request comes through an aggregator, the signature will be empty.
+
+A Wallet that supports aggregated signature should have the interface:
+```solidity
+interface IAggregatedWallet is IWallet {
+  function getAggregator() view returns (address);
+}
+
+```
+If the wallet supports aggregation, this method should return the aggress of the aggregator.
```

Comment:
aggress -> address

Reply:
what's bad about the aggress of the addregator ??!